openhtf/output/callbacks/__init__.py

```diff
@@ -82,6 +83,10 @@ def serialize_test_record(test_record):
   @staticmethod
   def open_file(filename):
     """Override method to alter file open behavior or file types."""
+    basepath = os.path.dirname(filename)
+
+    os.makedirs(basepath, exist_ok=True)
```

Comment:
note that `exists_ok` will bring PY2 compatibility, though I plan on doing that shortly anyway.